### PR TITLE
Test runner improvements

### PR DESCRIPTION
**Pull request by yhr**
Add some hooks to allow for arbitrary pre/post commands to be run for each tests and for copying rocksdb LOG files (useful for checking write stats in read-while-writing tests that does not include the write IOPS for example)

### CODE REVIEW

**Review by metaspace — APPROVED** [review]
:+1:

File: tests/run.sh [hunk 1]
```diff
@@ -62,6 +62,13 @@ do
     echo "$(tput setaf 1)FAILED$(tput sgr 0)"
     FAILED_TESTS=$((FAILED_TESTS+1))
   fi
+
+  if [ -v LOG_FILE_PATH ]; then
+    set -e
```
Comment (jsvaerke) [inline tests/run.sh:73]: Do you want to fail the test run if the copy fails? Also, -e is already active here, so you are switching to +e

File: tests/run.sh [hunk 2]
```diff
@@ -62,6 +68,19 @@ do
...
+    cp $LOG_FILE_PATH "$RESULT_PATH/${TEST/.sh/.LOG}"
+    set +e
+  fi
```
[the remark this answers is not in the document]
Comment (yhr) [reply tests/run.sh:76]: I considered that, but would like the output to be separated when running both write amp and log file monitoring per test

File: tests/run.sh (hunk 1 above)
Comment (yhr) [reply tests/run.sh:73]: will fix

**Review by jsvaerke — APPROVED** [review]

**Pull request merged**